### PR TITLE
Provisioning: fix for cannot save provisioned dashboard

### LINKED ISSUE #20211: Provisioning: Cannot save provisioned dashboard
**What happened**:
Starting grafana in development mode and see this error message each time provisioning running. 
```
DBUG[11-06|16:30:17] Start walking disk                       logger=provisioning.dashboard type=file name="gdev dashboards" path=devenv/dev-dashboards
DBUG[11-06|16:30:17] saving new dashboard                     logger=provisioning.dashboard type=file name="gdev dashboards" provisioner="gdev dashboards" file=/home/marcus/go/src/github.com/grafana/grafana/devenv/dev-dashboards/panel-polystat/polystat_test.json folderId=1985
EROR[11-06|16:30:17] failed to save dashboard                 logger=provisioning.dashboard type=file name="gdev dashboards" error="Cannot save provisioned dashboard"
```

**What you expected to happen**:
Provisioned dashboard to be saved.

**How to reproduce it (as minimally and precisely as possible)**:
This is an existing provisioned dashboard (1 year old).

dashboard_provisioning row:
```
id | dashboard_id | name | external_id | updated | check_sum -- | -- | -- | -- | -- | --
2562 | 2502 | gdev dashboards | /home/marcus/go/src/github.com/grafana/grafana/devenv/dev-dashboards/panel-polystat/polystat_test.json | 1573035122 | 977b8f242142d17578a4fe5098d64839
```

dashboard row:
```
id | version | slug | title | org_id | created | updated | updated_by | created_by | folder_id | uid -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
2502 | 5 | panel-tests-polystat | Panel Tests - Polystat | 1 | 2019-05-27 16:32:47 | 2019-11-06 10:12:02 | -1 | -1 | 1985 | Kp9Z0hTik
```

**Anything else we need to know?**:
Shouldn't validate argument always be false when provisioning saves dashboard? Now code looks like [this](https://github.com/grafana/grafana/blob/master/pkg/services/dashboards/dashboard_service.go#L192) and probably why it fails since allowUiUpdate for devenv provisioned dashboard in grafana repo don't have this flag set to true?

**Environment**:
- Grafana version: master

### PR DESCRIPTION
**Pull request by jongyllen**
**What this PR does / why we need it**:
There is a bug in provisioning causing dashboards from disk not being provisioned if `allowUiUpdates` is set to false

**Which issue(s) this PR fixes**:
Fixes #20211

### CODE REVIEW

**Review by torkelo — APPROVED** [review]

File: pkg/services/provisioning/dashboards/file_reader.go [hunk 1]
```diff
@@ -190,7 +190,7 @@ func (fr *fileReader) saveDashboard(path string, folderId int64, fileInfo os.Fil
 		CheckSum:   jsonFile.checkSum,
 	}
 
-	_, err = fr.dashboardProvisioningService.SaveProvisionedDashboard(dash, dp, fr.Cfg.AllowUiUpdates)
+	_, err = fr.dashboardProvisioningService.SaveProvisionedDashboard(dash, dp, true)
```
Comment (marefr) [inline pkg/services/provisioning/dashboards/file_reader.go]: I only see one usage of `SaveProvisionedDashboard` and that is here. I would remove the allowUiUpdate argument to `SaveProvisionedDashboard` alltogether and let the `SaveProvisionedDashboard` decide that provisioning validation not should happen.

**Review by marefr — CHANGES_REQUESTED** [review]
This works, but I suggest a change. See comment.

File: pkg/services/provisioning/dashboards/file_reader.go (hunk 1 above)
Comment (jongyllen) [reply pkg/services/provisioning/dashboards/file_reader.go]: Thanks, that is a great change. :)

**Review by marefr — APPROVED** [review]
Great :+1:

**Pull request merged**